src/couch_httpd_auth.erl

```diff
@@ -366,6 +366,19 @@ handle_session_req(#httpd{method='DELETE'}=Req, _AuthModule) ->
 handle_session_req(Req, _AuthModule) ->
     send_method_not_allowed(Req, "GET,HEAD,POST,DELETE").
 
+extract_username(Form) ->
+    try couch_util:get_value("name", Form) of
+    CouchFormat ->
+        case couch_util:get_value("username", Form, CouchFormat) of
+        undefined ->
+            throw({bad_request, <<"user accounts must have a username">>});
```

Comment:
I believe, "request payload must contain a username" is more helpful error message, since user don't create any account here, but tries to log in.